__tests__/app.test.js

```diff
@@ -1,82 +1,124 @@
-              })
-            expect(body).toEqual({
-                article_id: expect.any(Number),
-                title: expect.any(String),
-                topic: expect.any(String),
-                author: expect.any(String),
-                body: expect.any(String),
-                created_at: expect.any(String),
-                votes: expect.any(Number),
-                article_img_url: expect.any(String)
-            })
-        })
-    })
-    it('400 - responds with an error msg for an invalid article_id type', () => {
-        return request(app)
-        .get('/api/articles/not-a-number')
-        .expect(400)
-        .then(({body}) => {
-            expect(body.msg).toEqual('Invalid data type')
-        })
-    })
-})
\ No newline at end of file
+describe("GET /api", () => {
+  it("200 - should respond with an object", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual(endpoints);
+      });
+  });
+});
+
+describe("GET /api/articles/:articleId", () => {
+  it("200 - should respond with an object containing the correct article", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual({
+          article_id: expect.any(Number),
+          title: expect.any(String),
+          topic: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        });
+      });
+  });
+  it("400 - responds with an error msg for an invalid article_id type", () => {
+    return request(app)
+      .get("/api/articles/not-a-number")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("Invalid data type");
+      });
+  });
+  it("404 - responds with an error msg for an article_id that does not exist", () => {
+    return request(app)
+      .get("/api/articles/9999")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("Not found");
+      });
+  });
+});
+
+describe("GET /api/articles", () => {
+  it("200 - should respond with an array of articles", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        expect(Array.isArray(articles)).toBe(true);
+        expect(articles.length).toBe(13);
+        expect(body).toEqual({ articles: articles });
+      });
+  });
+  it("200 - should respond with an array of articles with the correct keys", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        if (articles.length === 0) {
```

Comment:
keep custom errors in the model function as a matter of good practise, having them in the test.js is a little messy